File: r/tests/testthat/test-reset.R
test_that("reset is exported", {
  expect_true("reset" %in% getNamespaceExports("bedrockbio"))
  expect_true(is.function(reset))
})

test_that("reset clears cached state", {
  pkg <- bedrockbio:::pkg
  pkg$manifest <- list("ns.tbl" = list())
  pkg$namespaces <- list(ns = list())
  bedrockbio:::reset()
  expect_null(pkg$manifest)
  expect_null(pkg$namespaces)
})

File: python/tests/test_reset.py
import bedrock_bio as bb
from bedrock_bio.config import config


def test_reset_is_exported():
    assert callable(bb.reset)
    assert "reset" in bb.__all__


def test_reset_clears_state():
    config.manifest = {"ns.tbl": {}}
    config.namespaces = {"ns": {}}
    bb.reset()
    assert config.manifest is None
    assert config.namespaces is None
